APA PsycNET.js

```diff
@@ -1,375 +1,277 @@
+						"something unusual (sigh). Please reload any APA page e.g. " +
+						"http://psycnet.apa.org/ in your browser and try again.");
+					Z.debug(data);
+				}
+			});
 		}
-	}
+	}, headers);
 }
 
-//retrieve RIS data
-//retry n times
-function fetchRIS(url, post, itemType, doc, retry) {
-	//language only available on the page;
-	var language = ZU.xpathText(doc, '//meta[@name="citation_language"]/@content');
-	ZU.doPost(url, post, function(text) {
-		//There's some cookie/session magic going on
-		//our first request for RIS might not succeed
-		var foundRIS = (text.indexOf('TY  - ') != -1);
-		if(!foundRIS && retry) {
-			//retry
-			Z.debug('No RIS data. Retrying (' + retry + ').');
-			fetchRIS(url, post, itemType, doc, --retry);
-			return;
-		} else if(!foundRIS) {
-			Z.debug('No RIS data. Falling back to scraping the page directly.');
-			scrapePage(doc, itemType);
-			return;
-		}
-		
-		//clean up (double) spacing in RIS
-		text = text.replace(/  +/g, ' ');
-		//Z.debug(text)
-		var translator = Zotero.loadTranslator("import");
-		translator.setTranslator("32d59d2d-b65a-4da4-b0a3-bdd3cfb979e7");
-
-		//Some authors are not entered according to spec
-		//We try to fix places where multiple authors are entered in one entry
-		text = text.replace(/^((?:A[U123]|ED) - )(.+?)$/mg,
-			function(match, tag, authors) {
-				var authorRE = /((?:[A-Z](?:\.?\s|\.))+)([-A-Za-z]+)/g;
-				var author, newStr='';
-				while(author = authorRE.exec(authors)) {
-					if(newStr) {	//not the first author
-						newStr += '\r\n';
-					}
-
-					newStr += tag + author[2] + ', ' + author[1].trim();
-				}
 
-				if(newStr) {
-					return newStr;
-				} else {	//we didn't find any authors
-					return match;
-				}
-			}
-		);
-			
-		text = text.replace(/^DESCRIPTORS -.+/gm, "");
-		
-		translator.setString(text);
-		translator.setHandler("itemDone", function(obj, item) {
-			//item.url = newurl;
-			
-			item.title = item.title.replace(/\.$/,'');
-			if (item.ISSN) {
-				var ISSN = item.ISSN.split(/\s*;\s*/);
-				var ISSNArray = [];
-				for (var i=0; i<ISSN.length; i++) {
-					ISSNArray.push(ZU.cleanISSN(ISSN[i]));
-				}
-				item.ISSN = ISSNArray.join(" ");
-				item.language = language;
-			}
-			finalizeItem(item, doc);		
+function processRIS(text, doc) {
+	var translator = Zotero.loadTranslator("import");
+	translator.setTranslator("32d59d2d-b65a-4da4-b0a3-bdd3cfb979e7");
+	translator.setString(text);
+	translator.setHandler("itemDone", function(obj, item) {
+		item.title = cleanTitle(item.title);
+		if (item.publication) item.publication = cleanTitle(item.publication);
+		if (item.bookTitle) item.bookTitle = cleanTitle(item.bookTitle);
+		if (item.series) item.series = cleanTitle(item.series);
+		if (item.place) item.place = item.place.replace(/\s+/g, ' ');
+		for (var i=0; i<item.tags.length; i++) {
+			item.tags[i] = item.tags[i].replace(/^\*/, '');
+		}
+		var pdfURL = attr(doc, 'a[href*="/fulltext"]');
```

Comment:
I noticed an open access article PDF is not being saved. Looks like attr() is missing the attr parameter, below works in browser at least:
``var pdfURL = attr(doc, 'a[href*="/fulltext"]','href');``

Reply:
This should be fixed in the newest commit. @tomellor Can you test it again?